design-system/react/src/components/CardList/CardList.tsx

```diff
@@ -0,0 +1,50 @@
+import { cx } from "@fuel-ui/css";
+import { createContext, useContext } from "react";
+
+import type { StackProps } from "..";
+import { Focus, Stack } from "..";
+import type { CreateComponent } from "../../utils";
+import { createComponent } from "../../utils";
+
+import { CardListItem } from "./CardListItem";
+
+type Context = {
+  isClickable?: boolean;
+};
+
+const ctx = createContext<Context>({} as Context);
+
+export function useCardListContext() {
+  return useContext(ctx);
+}
+
+export type CardListProps = StackProps & {
+  isClickable?: boolean;
+};
+
+export const CardList = createComponent<CardListProps>(
+  ({ children, className, isClickable, ...props }) => {
+    const classes = cx("fuel_card-list", className);
+    const content = (
+      <Stack {...props} className={classes}>
+        {children}
+      </Stack>
+    );
+    return (
+      <ctx.Provider value={{ isClickable }}>
+        {!isClickable ? (
+          content
+        ) : (
+          <Focus.ArrowNavigator asChild autoFocus>
+            {content}
+          </Focus.ArrowNavigator>
+        )}
+      </ctx.Provider>
+    );
+  }
+) as CreateComponent<CardListProps> & {
```

Comment:
We should be able to infer this via type and not force to use `as`.

createComponent<CardListProps & {
   id: string;
  Item: typeof CardListItem;
}>

Reply:
Can you create an issue and let this can be a technical debt maybe @luizstacio? I'll need to change so many things for this pull request to fix this. I'm using this pattern in a lot of components until now 🙄

Reply:
we can open an issue for that, but we should work on it soon, as the project is young, and this is not a good pattern that will be replicated not only here but on the projects that uses `fuel-ui`